uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkDay.php

```diff
@@ -165,27 +157,55 @@ private static function validateWorkHours($params) {
       );
     }
 
-    $hasTimesAndBreak = $timeFrom && $timeTo && $break;
-    if($isWorkingDay && !$hasTimesAndBreak) {
+    $hasAllRequiredFields = !is_null($timeFrom) && !is_null($timeTo) && !is_null($break) && !is_null($numberOfHours);
+    if($isWorkingDay && !$hasAllRequiredFields) {
       throw new InvalidWorkDayException(
-        'Time From, Time To and Break are required for Working Days'
+        'Time From, Time To, Break and Number of Hours are required for Working Days'
       );
     }
 
     $hasTimes = !is_null($timeFrom) && !is_null($timeTo);
     if($hasTimes && (strtotime($timeFrom) >= strtotime($timeTo))) {
       throw new InvalidWorkDayException(
-          'Time From should be less than Time To'
+        'Time From should be less than Time To'
       );
     }
 
+    if (!is_null($break)) {
+      if (!is_numeric($break)) {
+        throw new InvalidWorkDayException(
+          'Break should be a valid number'
+        );
+      } else if ($break < 0) {
+        throw new InvalidWorkDayException(
+          'Break cannot be less than 0'
+        );
+      }
+    }
+
     $secondsInWorkingHours = strtotime($timeTo) - strtotime($timeFrom);
     $secondsInBreak = $break * 3600;
     if($hasTimes && ($secondsInBreak >= $secondsInWorkingHours)) {
       throw new InvalidWorkDayException(
-          'Break should be less than the number of hours between Time From and Time To'
+        'Break should be less than the number of hours between Time From and Time To'
       );
     }
+
+    if (!is_null($numberOfHours)) {
+      if (!is_numeric($numberOfHours)) {
+        throw new InvalidWorkDayException(
+          'Number of Hours should be a valid number'
+        );
+      } else if ($numberOfHours <= 0 || $numberOfHours >= 24) {
+        throw new InvalidWorkDayException(
+          'Number of Hours should be between 0 and 24'
+        );
+      } else if (strpos($numberOfHours / 0.25, '.' )) {
+        throw new InvalidWorkDayException(
+          'Number of Hours should be divisible by 0.25'
```

Comment:
What about a more user friendly error message? 
e.g `Number of Hours are only allowed in 15 minutes blocks`

Reply:
Done.